tests/aws/services/events/test_events.py

```diff
@@ -1000,3 +1084,259 @@ def _get_sqs_messages():
         with pytest.raises(ClientError) as e:
             aws_client.events.describe_event_bus(Name=nonexistent_event_bus)
         snapshot.match("non-existent-bus", e.value.response)
+
+
+class TestEventRule:
+    @markers.aws.validated
+    @pytest.mark.parametrize("bus_name", ["custom", "default"])
+    def test_put_list_with_prefix_describe_delete_rule(
+        self, bus_name, create_event_bus, put_rule, aws_client, snapshot
+    ):
+        if bus_name == "custom":
+            bus_name = f"bus-{short_uid()}"
+            snapshot.add_transformer(snapshot.transform.regex(bus_name, "<bus-name>"))
+            create_event_bus(Name=bus_name)
+
+        rule_name = f"test-rule-{short_uid()}"
+        snapshot.add_transformer(snapshot.transform.regex(rule_name, "<rule-name>"))
+        response = put_rule(
+            Name=rule_name,
+            EventPattern=json.dumps(TEST_EVENT_PATTERN),
+            EventBusName=bus_name,
+        )
+        snapshot.match("put-rule", response)
+
+        # NamePrefix required for default bus against AWS
+        response = aws_client.events.list_rules(NamePrefix=rule_name, EventBusName=bus_name)
+        snapshot.match("list-rules", response)
+
+        response = aws_client.events.describe_rule(Name=rule_name, EventBusName=bus_name)
+        snapshot.match("describe-rule", response)
+
+        response = aws_client.events.delete_rule(Name=rule_name, EventBusName=bus_name)
+        snapshot.match("delete-rule", response)
+
+        response = aws_client.events.list_rules(NamePrefix=rule_name, EventBusName=bus_name)
+        snapshot.match("list-rules-after-delete", response)
+
+    @markers.aws.validated
+    def test_put_multiple_rules_with_same_name(self, put_rule, aws_client, snapshot):
+        rule_name = f"test-rule-{short_uid()}"
+        snapshot.add_transformer(snapshot.transform.regex(rule_name, "<rule-name>"))
+
+        response = put_rule(
+            Name=rule_name,
+            EventPattern=json.dumps(TEST_EVENT_PATTERN),
+        )
+        snapshot.match("put-rule", response)
+
+        if is_aws_cloud():
+            time.sleep(10)
```

Comment:
This can be flaky and non-deterministic. An alternative is to retry until it succeeds (with a timeout resulting in test failure)

Reply:
it is not really required, removed in: https://github.com/localstack/localstack/pull/10613/commits/eb01e64cb1f2540cc834fa616f7bd2a546f8356c

polling for a change condition would be unnecessary since if the put rule failed the snapshot will fail